onnxruntime/core/graph/runtime_optimization_record.h

```diff
@@ -75,21 +76,19 @@
   }
 };
 
-struct NodeIndexAndKernelDefHash {
-  NodeIndex node_index;
-  HashValue kernel_def_hash;
-};
-
 /** Information for a single runtime optimization.
 It does not contain information about the optimizer itself, that should be maintained seperately.
 */
 struct RuntimeOptimizationRecord {
   /** The optimization action identifier. */
   std::string action_id;
+
   /** The nodes to consider for optimization. */
   NodesToOptimizeIndices nodes_to_optimize_indices;
-  /** Any new nodes introduced by the optimization. */
-  std::vector<NodeIndexAndKernelDefHash> produced_nodes;
+
+  using ProducedOpIdVector = InlinedVector<OpIdentifier, 1>;
```

Comment:
>, 1>; [](http://example.com/codeflow?start=55&length=5)

Would std::array be a better container?

Reply:
it is expected to be a single element now, but potentially we could have optimizations that produce more than one node.